agent/api/task/task_test.go

```diff
@@ -2816,3 +2816,129 @@ func TestAssociationByTypeAndName(t *testing.T) {
 	association, ok = task.AssociationByTypeAndName("other-type", "dev1")
 	assert.False(t, ok)
 }
+
```

Comment:
You could think about using table driven tests for some of these

Reply:
I have reduced the number of tests by combining the cases in one...It brought down the number of tests to 1 and also checking all scenarios.

Reply:
consider making the container names mirror the use case of the test.